modules/light-clients/08-wasm/types/msgs.go

```diff
@@ -33,3 +35,73 @@ func (m MsgStoreCode) GetSigners() []sdk.AccAddress {
 	}
 	return []sdk.AccAddress{signer}
 }
+
+type QueryMsg struct {
+	Status               *statusMsg               `json:"status,omitempty"`
+	ExportMetadata       *exportMetadataMsg       `json:"export_metadata,omitempty"`
+	TimestampAtHeight    *timestampAtHeightMsg    `json:"timestamp_at_height,omitempty"`
+	VerifyClientMessage  *verifyClientMessageMsg  `json:"verify_client_message,omitempty"`
+	VerifyMembership     *verifyMembershipMsg     `json:"verify_membership,omitempty"`
+	VerifyNonMembership  *verifyNonMembershipMsg  `json:"verify_non_membership,omitempty"`
+	CheckForMisbehaviour *checkForMisbehaviourMsg `json:"check_for_misbehaviour,omitempty"`
+}
+
+type (
+	statusMsg            struct{}
+	exportMetadataMsg    struct{}
+	timestampAtHeightMsg struct {
+		Height exported.Height `json:"height"`
+	}
+	verifyClientMessageMsg struct {
+		Header       *Header       `json:"header,omitempty"`
+		Misbehaviour *Misbehaviour `json:"misbehaviour,omitempty"`
+	}
+	verifyMembershipMsg struct {
+		Height           exported.Height `json:"height"`
+		DelayTimePeriod  uint64          `json:"delay_time_period"`
+		DelayBlockPeriod uint64          `json:"delay_block_period"`
+		Proof            []byte          `json:"proof"`
+		Path             exported.Path   `json:"path"`
+		Value            []byte          `json:"value"`
+	}
+	verifyNonMembershipMsg struct {
+		Height           exported.Height `json:"height"`
+		DelayTimePeriod  uint64          `json:"delay_time_period"`
+		DelayBlockPeriod uint64          `json:"delay_block_period"`
+		Proof            []byte          `json:"proof"`
+		Path             exported.Path   `json:"path"`
+	}
+	checkForMisbehaviourMsg struct {
+		Header       *Header       `json:"header,omitempty"`
+		Misbehaviour *Misbehaviour `json:"misbehaviour,omitempty"`
+	}
+)
+
+type SudoMsg struct {
+	Initialize                    *initializeMsg                    `json:"initialize,omitempty"`
+	UpdateState                   *updateStateMsg                   `json:"update_state,omitempty"`
+	UpdateStateOnMisbehaviour     *updateStateOnMisbehaviourMsg     `json:"update_state_on_misbehaviour,omitempty"`
+	VerifyUpgradeAndUpdateState   *verifyUpgradeAndUpdateStateMsg   `json:"verify_upgrade_and_update_state,omitempty"`
+	CheckSubstituteAndUpdateState *checkSubstituteAndUpdateStateMsg `json:"check_substitute_and_update_state,omitempty"`
+}
+
+type (
+	initializeMsg struct {
+		ClientState    *ClientState    `json:"client_state"`
+		ConsensusState *ConsensusState `json:"consensus_state"`
+	}
+	updateStateMsg struct {
+		Header *Header `json:"header,omitempty"`
```

Comment:
I changed this from the [original message](https://github.com/cosmos/ibc-go/blob/feat/wasm-clients/modules/light-clients/08-wasm/types/update.go#L53-L55), since in theory for update state you only use the header and not misbehaviour. But this change also requires changes in the contracts, so I need to check with them.